odoo/addons/base/models/res_partner.py

```diff
@@ -475,6 +475,11 @@ def _check_parent_id(self):
         if self._has_cycle():
             raise ValidationError(_('You cannot create recursive Partner hierarchies.'))
 
+    @api.constrains('company_id')
+    def _check_commercial_parnter_company(self):
+        if self.company_id and self.commercial_partner_id.company_id and self.company_id != self.commercial_partner_id.company_id:
```

Comment:
A global partner with its parent limited to a company will cause access errors, no?
```suggestion
        if self.commercial_partner_id.company_id and self.company_id != self.commercial_partner_id.company_id:
```

Reply:
So, I pushed a check that only verifies if both companies are set and they are different. I think `company=False` on any would work fine in terms of access...